controllers/machineset_controller.go

```diff
@@ -219,25 +218,73 @@ func (r *MachinesetReconciler) updateLabelsInNode(ctx context.Context, node *cor
 	return nil
 }
 
-func (r MachinesetReconciler) updateTaintsInNode(ctx context.Context, machine *machinev1.Machine, node *corev1.Node) error {
+// updateTaintsInNode ensures all taints on a node are expected. Expected taints on a node are any taints specified
+// on a machine as well as any other NoSchedule taints.
+func (r *MachinesetReconciler) updateTaintsInNode(ctx context.Context, machine *machinev1beta1.Machine, node *corev1.Node) error {
+	noScheduleTaint := &corev1.Taint{
+		Key:    corev1.TaintNodeUnschedulable,
```

Comment:
I wonder if we want to add _all_ corev1 taints here, and not just NodeUnscheduleable (also memory pressure, etc).  Doesn't have to be done as part of this work, but would be a good "intro bug" for a maintenance sprint or something.